lib/carousel_slider.dart

```diff
@@ -22,17 +22,65 @@ class CarouselSlider extends StatefulWidget {
       this.pauseAutoPlayOnTouch,
       this.enlargeCenterPage = false,
       this.onPageChanged,
-        this.scrollPhysics,
-      this.scrollDirection: Axis.horizontal})
-      : this.realPage = enableInfiniteScroll ? realPage + initialPage : initialPage,
+      this.scrollPhysics,
+      this.scrollDirection: Axis.horizontal,
+      this.onWidgetCreated})
+      : this.realPage =
+            enableInfiniteScroll ? realPage + initialPage : initialPage,
+        this.itemCount = items.length,
+        this.itemBuilder = null,
         this.pageController = PageController(
           viewportFraction: viewportFraction,
-          initialPage: enableInfiniteScroll ? realPage + initialPage : initialPage,
-        );
+          initialPage:
+              enableInfiniteScroll ? realPage + initialPage : initialPage,
+        ) {
+    if (onWidgetCreated != null) onWidgetCreated(this);
```

Comment:
Why did you add this logic?

Reply:
i need this callback to take access to 'pageController' then i use them to slide items by ui buttons

Reply:
It's unnecessary callback cause you always can do this by following simple code:

```
@override
  Widget build(BuildContext context) {
    final carousel = CarouselSlider(...);
    final controller = carousel.pageController;

    return carousel;
  }
```